exercises/say/src/example.c

```diff
@@ -0,0 +1,148 @@
+#include "say.h"
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <math.h>
+
+int say(int64_t input, char **ans)
+{
+   if (input < 0 || input > 999999999999 || ans == NULL)
+      return -1;
+
+   /* 'zero' is a corner case */
+   if (input == 0) {
+      *ans = malloc(5);
+      if (*ans == NULL) {
+         fprintf(stderr, "Memory error!/n");
+         return -1;
+      }
+      strcpy(*ans, "zero");
+      return 0;
+   }
+
+   const char *below_10[] = {
+      "", "one", "two", "three", "four",
+      "five", "six", "seven", "eight", "nine"
+   };
+   const int below_10_len[] = {
+      0, 3, 3, 5, 4,
+      4, 3, 5, 5, 4
+   };
+
+   const char *below_20[] = {
+      "ten", "eleven", "twelve", "thirteen", "fourteen",
+      "fifteen", "sixteen", "seventeen", "eighteen", "nineteen"
+   };
+   const int below_20_len[] = {
+      3, 6, 6, 8, 8,
+      7, 7, 9, 8, 8
+   };
+
+   const char *below_100[] = {
+      "", "", "twenty", "thirty", "forty",
+      "fifty", "sixty", "seventy", "eighty", "ninety"
+   };
+   const int below_100_len[] = {
```

Comment:
These arrays may be a good candidate for introducing a typedef'd struct that holds both the number and the word. What do you think?

It may even be a good opportunity for you to learn what X-macros are and what they are often used for, if you haven't come across them yet. Not suggesting that X-macros should be introduced to the C track here though as they would likely be an exercise (or several) in themselves 😃

Reply:
Also, the arrays could have file-scope, rather than being part of the function.

Reply:
### X macros info if interested:
- [The late, great, Dr. Dobbs](http://www.drdobbs.com/the-new-c-x-macros/184401387)
- [embedded.com part 1](https://www.embedded.com/design/programming-languages-and-tools/4403953/C-language-coding-errors-with-X-macros-Part-1)
- [embedded.con part 2](https://www.embedded.com/design/programming-languages-and-tools/4405283/Reduce-C--language-coding-errors-with-X-macros---Part-2)
- [embedded.com part 3](https://www.embedded.com/design/programming-languages-and-tools/4408127/Reduce-C-language-coding-errors-with-X-macros--Part-3)

Reply:
It is indeed a lot of clutter like this :/
I have to admit I never thought about struct on this one, which is obviously a better, more integrated choice.
Also never heard of the 'X-macros'. It looks interesting - thanks for the links! Always glad to learn new stuff.

Reply:
For this exercise I decided to go the 'struct'-route.
As you said maybe would be nice to have exercises with X-macros. I personally will gladly experiment more with them.

Reply:
I use X-macros a lot at work. They add some cognitive overhead where they are used, but if you use them at the right time they are incredibly powerful.